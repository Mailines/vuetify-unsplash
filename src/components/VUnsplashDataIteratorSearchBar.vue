<template>
  <v-card
    class="mx-0 mx-sm-6 mt-4 mb-2 px-4 px-sm-8 py-2 rounded-pill d-flex align-center"
  >
    <v-container fluid>
      <v-layout row>
        <v-flex xs12 class="d-flex justify-center align-center ">
          <v-text-field
            class="d-flex align-self-center mb-n8 v-unplash-search-bar-layout"
            v-model="searchText"
            prepend-icon="mdi-magnify"
            clear-icon="mdi-close"
            clearable
            outlined
            :color="accentColor"
            :label="searchTextFieldPlaceholder"
            type="text"
            :disabled="searchLoading"
            :loading="searchLoading"
            @keydown.esc.prevent="closeDialog"
            @keydown.enter.prevent="search(searchText)"
          >
            <template v-slot:append-outer>
              <div
                class="append-outer-vunsplash d-flex justify-center align-center pl-2"
              >
                <v-progress-circular
                  v-if="searchLoading"
                  :color="accentColor"
                  class="ml-3 mt-1"
                  indeterminate
                  :width="3"
                  size="24"
                ></v-progress-circular>
                <v-btn
                  v-else-if="searchText"
                  icon
                  :color="accentColor"
                  @click="search(searchText)"
                >
                  <v-icon>mdi-send</v-icon>
                </v-btn>
              </div>
            </template>
          </v-text-field>
        </v-flex>
      </v-layout>
    </v-container>
  </v-card>
</template>
<script>
export default {
  props: {
    accentColor: {
      type: String,
      required: true
    },
    searchLoading: {
      type: Boolean,
      required: true
    },
    searchTextFieldPlaceholder: {
      type: String,
      required: true
    }
  },
  data() {
    return {
      searchText: null
    };
  },
  beforeDestroy() {
    this.searchText = null;
  },
  methods: {
    closeDialog() {
      this.$emit("close");
    },
    search(searchText) {
      this.$emit("search", searchText);
    }
  }
};
</script>
<style scoped>
.append-outer-vunsplash {
  margin-top: -8px !important;
}
</style>
<style>
.v-unplash-search-bar-layout > .v-input__prepend-outer {
  margin-top: 16px !important;
}
</style>
