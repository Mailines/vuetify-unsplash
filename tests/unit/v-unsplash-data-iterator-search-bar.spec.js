import VUnsplashDataIteratorSearchBar from "../../src/components/VUnsplashDataIteratorSearchBar";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashDataIteratorSearchBar.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashDataIteratorSearchBar, {
      localVue,
      vuetify,
      propsData: {
        searchLoading: false,
        searchTextFieldPlaceholder: "Search...",
        accentColor: "#1976d2"
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.destroy();
  });
  it("emits close event when calling closeDialog()", async () => {
    wrapper.vm.closeDialog();
    expect(wrapper.emitted().close).toBeTruthy();
  });
  it("emits search event with text when calling search(searchText)", async () => {
    const searchText = "testSearch";
    wrapper.vm.search(searchText);
    expect(wrapper.emitted().search[0][0]).toBe(searchText);
  });
});
