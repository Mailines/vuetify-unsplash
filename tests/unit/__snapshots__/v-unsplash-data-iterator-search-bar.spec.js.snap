// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashDataIteratorSearchBar.vue renders and match snapshot 1`] = `
<div class="mx-0 mx-sm-6 mt-4 mb-2 px-4 px-sm-8 py-2 rounded-pill d-flex align-center v-card v-sheet theme--light">
  <div class="container container--fluid">
    <div class="layout row">
      <div class="flex d-flex justify-center align-center xs12">
        <div class="v-input d-flex align-self-center mb-n8 v-unplash-search-bar-layout theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer">
            <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate mdi mdi-magnify theme--light"></i></div>
          </div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span>â€‹</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-4" class="v-label theme--light" style="left: 0px; position: absolute;">Search...</label><input id="input-4" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--clear"><button disabled="disabled" type="button" aria-label="clear icon" class="v-icon notranslate v-icon--disabled v-icon--link mdi mdi-close theme--light"></button></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
          <div class="v-input__append-outer">
            <div class="append-outer-vunsplash d-flex justify-center align-center pl-2">
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
